### DISCUSSION TITLE
How to set a user for GitPoller or GitStep?

### DISCUSSION THREAD

**Discussion by dimavs** [buildbot/buildbot#6351] [Q&A]
Buildbot worker service is running under a buildbot user. Is it possible to pass a different user to GitPoller or GitStep (something like 'ssh -l foo')?
The only way I can change user is just to set it in .ssh/config, it would be nice to do it from buildbot scripts.

**Comment by p12tic** [comment]
Can you use `ssh://<temp_user>@<host>/<repo_path>` as the remote?

**Reply by dimavs** [reply]
Yes it works for the repository itself, but submodules would fail.

**Reply by p12tic** [reply]
Does `GIT_SSH_COMMAND="ssh -l user"` work? This could be a solution to include in Buildbot.

**Reply by dimavs** [reply]
Hm, thanks, I'll try it again. I tried it first but I had some issues passing environment variables to Git step (apparently environment is passed only to ShellCommand steps).

**Reply by p12tic** [reply]
I'm more asking about whether this works using plain git. If it works, great, we can modify the Git step to include it.

**Reply by dimavs** [reply]
Yes, it works fine: 

```
# sudo -u buildbot git -c 'core.sshCommand=ssh -l build -i ./id -o UserKnownHostsFile=./kh -o IdentitiesOnly=yes' ls-remote <repo>
# sudo -u buildbot git -c 'core.sshCommand=ssh -l build -i <absolute path to id file> -o UserKnownHostsFile=<absolute path to known hosts> -o IdentitiesOnly=yes' clone --recurse-submodules <repo> <dir>
```
or just _git clone_ and git _sibmodule update --init --recursive_ also work fine
